### ISSUE TITLE
select2_from_ajax not working when editing existing record

### ISSUE THREAD

**Issue by slickorange** [Laravel-Backpack/CRUD#1794]
# Bug report
My select2_from_ajax fields are not working when editing an existing record

### What I did:
I typed something into a select2_from_ajax field on an existing record

### What I expected to happen:
Results should load. (It loads on a new record)

### What happened:
Results do not load. I get a network error in Chrome devtools but the response is completely empty.

### What I've already tried to fix it:
I tried the api URL that the field uses directly (with the same query) and everything loads as expected.
I see that the request includes all form data, including one of my text fields with a big paragraph. When I take out this paragraph it works. 

How would I fix this issue where other field's content prevent the ajax field from loading?

Example Request:
`https://myapp.com/admin/api/client?q=o&form%5B0%5D%5Bname%5D=_token&form%5B0%5D%5Bvalue%5D=MfnEqRHzfB84dhLWlbt0a7TwcKScJ5QZ0w14ufz4&form%5B1%5D%5Bname%5D=_method&form%5B1%5D%5Bvalue%5D=PUT&form%5B2%5D%5Bname%5D=http_referrer&form%5B2%5D%5Bvalue%5D=https%3A%2F%2Fapp.generationindex.com%2Fadmin%2Fassessment&form%5B3%5D%5Bname%5D=id&form%5B3%5D%5Bvalue%5D=2&form%5B4%5D%5Bname%5D=code&form%5B4%5D%5Bvalue%5D=5c35893f71edd&form%5B5%5D%5Bname%5D=template_id&form%5B5%5D%5Bvalue%5D=1&form%5B6%5D%5Bname%5D=departments&form%5B6%5D%5Bvalue%5D=%5B%7B%22label%22%3A%22Top%20Management%22%2C%22value%22%3A%22top_management%22%7D%2C%7B%22label%22%3A%22Finance%22%2C%22value%22%3A%22finance%22%7D%5D&form%5B7%5D%5Bname%5D=intro&form%5B7%5D%5Bvalue%5D=%3Cdiv%3EThe%20aim%20of%20this%20assessment%20is%20to%20provide%20the%20leadership%20team%20of%20Hirthe%20PLC%20an%20opportunity%20to%20assess%20the%20working%20experience%20of%20its%20personnel.%3C%2Fdiv%3E%0D%0A%3Cdiv%3EPlease%20take%20note%20that%20%3Cstrong%3ENO%3C%2Fstrong%3E%20personal%20information%20is%20collected%20during%20this%20survey%2C%20and%20are%20you%20encouraged%20to%20be%20as%20true%20as%20possible%20in%20your%20response%20to%20the%20questions%20asked.%26nbsp%3B%3Cbr%20%2F%3E%3Cbr%20%2F%3EDuring%20this%20assessment%20four(4)%20areas%20of%20Organisational%26nbsp%3BCulture%20will%20be%20assessed%3A%3Cbr%20%2F%3E%26nbsp%3B-%20Culture%20of%20Community%20(ability%20to%20work%20as%20a%20team)%3Cbr%20%2F%3E%26nbsp%3B-%20Culture%20towards%20Personal%20Growth%3Cbr%20%2F%3E%26nbsp%3B-%20Culture%20of%20Change%26nbsp%3B%3Cbr%20%2F%3E%26nbsp%3B-%20Culture%20of%20Leadership%20Accessibility%20(organisational%20structure)%3C%2Fdiv%3E%3Cbr%3E%0D%0A%3Cdiv%3EThe%20purpose%20of%20the%20assessment%20is%20to%20gain%20a%20deeper%20understanding%20of%20the%20current%20work%20environment%20within%20Hirthe%20PLC%2C%20and%20to%20what%20extend%20this%20environment%20allows%20for%20a%20healthy%20multi-generational%20work%20environment.%3C%2Fdiv%3E%0D%0A%3Cdiv%3EIt%20is%20therefore%20pivotal%20to%20ensure%20that%20you%20have%20read%20thoroughly%20through%20each%20question%20before%20responding.%20There%20are%20no%20correct%20nor%20wrong%20answers%2C%20but%20merely%20you%20expressing%20your%20personal%20experience%20through%20the%20questions%20asked.%26nbsp%3B%3C%2Fdiv%3E%0D%0A%3Cdiv%3EEstimated%20time%20required%20to%20complete%20the%20Generational%20Engagement%20Index%20assessment%3A%207%20minutes.%26nbsp%3B%3C%2Fdiv%3E&form%5B8%5D%5Bname%5D=survey_header_color&form%5B8%5D%5Bvalue%5D=%23dd4b39&form%5B9%5D%5Bname%5D=survey_text_color&form%5B9%5D%5Bvalue%5D=%23ffffff&form%5B10%5D%5Bname%5D=report_overview_text&form%5B10%5D%5Bvalue%5D=%3Cp%3EThroughout%20the%20years%20scholars%20have%20researched%20organisational%20patterns%2C%20behaviour%20and%20culture%2C%20all%20to%20determine%20how%20and%20why%20they%20are%20going%20through%20change.%20With%20many%20partial%20explanations%20available%20on%20the%20life%20in%20organisations%2C%20the%20author%20of%20%3Cem%3E%E2%80%98Strategic%20Management%20and%20Organisational%20Dynamics%E2%80%99%3C%2Fem%3E%2C%20Ralph%20D.%20Stacey%2C%20makes%20it%20clear%20that%20there%20is%20no%20singular%20explanation%20as%20to%20why%20organisations%20change%2C%20just%20many%20theories%20based%20on%20its%20own%20unique%20experiences.%20However%20according%20to%20the%20book%20%3Cem%3E%E2%80%98Business%20Restructuring%3A%20An%20Action%20Template%20for%20Reducing%20Cost%20and%20Growing%20Profit%E2%80%99%3C%2Fem%3E%2C%20written%20by%20C.%20Zilka%2C%20an%20organisation%20needs%20to%20regularly%20look%20at%20itself%20and%20evaluate%20its%20effectiveness%20with%20that%20which%20is%20happening%20in%20a%20broader%20context.%20It%20needs%20to%20look%20at%20the%20impact%20and%20influence%20of%20the%20external%20environment%2C%20and%20how%20it%20needs%20to%20adjust%20to%20it.%3Cbr%3E%20%3C%2Fp%3E%0D%0A%20%20%20%20%20%20%20%20%3Cp%3E%3Cem%3E%E2%80%9CEvery%20company%2C%20regardless%20of%20size%20or%20product%20offering%2C%20should%20take%20a%20hard%20look%20at%20itself%20every%20year%20using%20rigorous%20self-assessment%20to%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20determine%20what%20is%20working%20in%20its%20organisation%20as%20well%20as%20what%20is%20not.%20It%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20is%20also%20essential%20to%20understand%20the%20external%20environment%20and%20howd%20impacts%20your%20business%20today%20as%20well%20as%20in%20the%20future.%E2%80%9D%20%3C%2Fem%3E%20%20%20%20%20%20%20%20%3C%2Fp%3E%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Cp%3EAcknowledging%20that%20there%20are%20many%20varieties%20of%20assessments%20available%20online%2C%20this%20survey%20served%20to%20assist%20%3Cstrong%3EHirthe%20PLC%3C%2Fstrong%3E%20in%20analysing%20its%20effectiveness%20and%20openness%20to%20positively%20attract%20and%20accommodate%20the%20younger%20leadership%20generation%20of%20today.%20Our%20research%20has%20revealed%2C%20amongst%20many%20things%2C%20four(4)%20key%20cultural%20requirements%20for%20an%20younger%20leadership%20generation%20to%20engage%20and%20partake%20longterm%20in%20the%20mission-vision%20of%20an%20organisation.%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20Although%20there%20seems%20to%20be%20a%20good%20balance%20between%20the%20four(4)%20areas%20of%20assessment%2C%20the%20feedback%20received%20also%20indicates%20that%20%3Cstrong%3EHirthe%20PLC%3C%2Fstrong%3E%20has%20a%20positive%20environment%20that%20provides%20excellent%20opportunities%20for%20Team%20Dynamics.%20This%20could%20be%2C%20but%20not%20limited%20to%2C%20due%20to%20the%20openness%20and%20willingness%20of%20staff%20to%20respond%20when%20someone%20requires%20assistance%2C%20as%20well%20as%20the%20willingness%20from%20leadership%20to%20regularly%20ask%20for%20feedback%20on%20challenges%20faced%20within%20the%20organisation.%20%3C%2Fp%3E&form%5B11%5D%5Bname%5D=current_tab&form%5B11%5D%5Bvalue%5D=assessment&form%5B12%5D%5Bname%5D=save_action&form%5B12%5D%5Bvalue%5D=save_and_back`

### Backpack, Laravel, PHP, DB version:
CRUD: 3.5,,
Laravel: 5.6
PHP: 7.3
MySQL 14.14

**Comment by imansugirman** [comment]
Can u share Route and Controller and Controller location. because your url https://myapp.com/admin/api/client?. maybe wrong.

Try 

``` 
https://myapp.com/api/client
```

**Comment by tabacitu** [comment]
Hi @slickorange ,

If you don't have chained select2s (one select depends on another select) you can use the (yet undocumented) ```select2_from_ajax``` parameter:
```
	    'include_all_form_fields'  => false, // optional - only send the current field through AJAX (for a smaller payload if you're not using multiple chained select2s)
```

Adding this to your addField() definition will make sure you only send the current field through AJAX. That way the big text won't be sent, and you won't have this problem.

I've just added docs for it - it's available in CRUD 3.6.

Hope it helps.
Cheers!

**Comment by amis5745-hub** [comment]
https://github.com/Laravel-Backpack/CRUD/issues/1794
